users/models.py

```diff
@@ -468,6 +482,21 @@ class ServiceSubscription(models.Model):
         null=True
     )
 
+    reference_number = models.BigIntegerField(
+        blank=True,
+        null=True,
+        unique=True,
+        verbose_name=_('Reference number for paying for this service subscription'),
+        help_text=_('Pay for this service with this reference number'),
+    )
+
+    warning_sent = models.BooleanField(
+        blank=False,
+        null=False,
+        default=False,
+        help_text=_('Set to True when a expiration warning message has been sent to user.')
+    )
```

Comment:
suggested name change: "reminder_sent" and perhaps a datetime datetime or even better a list of dates when reminders have been sent. this way we can say "this is the third and final reminder"